baselines/asset/src/v1/asset_service_client.ts.baseline

```diff
@@ -73,40 +73,34 @@ export class AssetServiceClient {
    *     your project ID will be detected automatically.
    * @param {string} [options.apiEndpoint] - The domain name of the
    *     API remote host.
+   * @param {gax.ClientConfig} [options.clientConfig] - client configuration override.
+   *     Client configuration set in the JSON config
+   *     (`src/v1/asset_service_client_config.json`)
+   *     can be overridden by passing `clientConfig` to the client constructor.
+   *     The detailed structure of the clientConfig can be found
+   *     {@link https://github.com/googleapis/gax-nodejs/blob/master/src/gax.ts#L564-L611 here}.
+   *     Pass any subset of that JSON config as `options.clientConfig`.
+   * @param {boolean} fallback - Use HTTP fallback mode.
+   *     In fallback mode, a special browser-compatible transport implementation is used
+   *     instead of gRPC transport. In browser context (if the `window` object is defined)
+   *     the fallback mode is enabled automatically; set `options.fallback` to `false`
+   *     if you need to override this behavior.
    */
 
   constructor(opts?: ClientOptions) {
-    // Ensure that options include the service address and port.
+    // Ensure that options include all the required fields.
     const staticMembers = this.constructor as typeof AssetServiceClient;
-    const servicePath = opts && opts.servicePath ?
-        opts.servicePath :
-        ((opts && opts.apiEndpoint) ? opts.apiEndpoint :
-                                      staticMembers.servicePath);
-    const port = opts && opts.port ? opts.port : staticMembers.port;
-
-    if (!opts) {
-      opts = {servicePath, port};
-    }
-    opts.servicePath = opts.servicePath || servicePath;
-    opts.port = opts.port || port;
-
-    // users can override the config from client side, like retry codes name.
-    // The detailed structure of the clientConfig can be found here: https://github.com/googleapis/gax-nodejs/blob/master/src/gax.ts#L546
-    // The way to override client config for Showcase API:
-    //
-    // const customConfig = {"interfaces": {"google.showcase.v1beta1.Echo": {"methods": {"Echo": {"retry_codes_name": "idempotent", "retry_params_name": "default"}}}}}
-    // const showcaseClient = new showcaseClient({ projectId, customConfig });
-    opts.clientConfig = opts.clientConfig || {};
-
-    // If we're running in browser, it's OK to omit `fallback` since
-    // google-gax has `browser` field in its `package.json`.
-    // For Electron (which does not respect `browser` field),
-    // pass `{fallback: true}` to the AssetServiceClient constructor.
+    const servicePath = opts?.servicePath ?? opts?.apiEndpoint ?? staticMembers.servicePath;
```

Comment:
I'm being wildly pedantic, but this could be considered a breaking change.  `??` has subtle different behavior from `||` in the way they treat falsey non-null/undefined values.  In practice, this means if the `opts.servicePath` was set to `''`, this would be a behavioral change.  I defer to @bcoe to tell us how much this matters.

Reply:
It should be safe to use `||` here, right?  if I write `opts?.servicePath || opts?.apiEndpoint || staticMembers.servicePath`, it will be the same as before, but still much more readable. WDYT?

Reply:
I tend to agree.

Reply:
I thinking the optional chain with the || is a fine compromise. I feel like someone might expect an empty string to default to the API endpoint.

Reply:
Done!